sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -92,6 +92,29 @@ object CosmosContainerConfig {
   }
 }
 
+case class CosmosReadConfig(inferSchemaSamplingSize: Int, inferSchemaEnabled: Boolean)
+
+object CosmosReadConfig {
+    private val DefaultSampleSize: Int = 1000
+
+    val inferSchemaSamplingSize = CosmosConfigEntry[Int](key = "spark.cosmos.read.inferSchemaSamplingSize",
+        mandatory = false,
+        parseFromStringFunction = size => size.toInt,
+        helpMessage = "Sampling size to use when inferring schema")
+
+    val inferSchemaEnabled = CosmosConfigEntry[Boolean](key = "spark.cosmos.read.inferSchemaEnabled",
+        mandatory = false,
+        parseFromStringFunction = enabled => enabled.toBoolean,
+        helpMessage = "Whether schema inference is enabled or should return raw json")
+
+    def parseCosmosReadConfig(cfg: Map[String, String]): CosmosReadConfig = {
+        val samplingSize = CosmosConfigEntry.parse(cfg, inferSchemaSamplingSize)
+        val enabled = CosmosConfigEntry.parse(cfg, inferSchemaEnabled)
+
+        CosmosReadConfig(samplingSize.getOrElse(DefaultSampleSize), enabled.getOrElse(true))
```

Comment:
I think enabled should have a default value of "false" - and I also think you still need to wire up the CosmosItemDataSource.inferSchema to return the same schema

I think it would be risky to apply an effective schema to the Table created by the TableProvider with a different schema than returned from the TableProvider.inferSchema call. For writes we can basically tell Spark to not validate the output of inferSchema with the ACCEPT_ANY_SCHEMA capability (which is fine - as long as you return the default schema when the spark.cosmos.read.inferSchemaEnabled" property isn't set this would simply work for writes. Worst case when customers set the property even on write it will give a small perf hit. By disabling it by default on reads you also only add the perf hit if explicitly requested.
My concern is that the TableProvider.inferschema is most likely called early by the catalyst infrastructure to generate the query plan - so if we return inconsistent schema it would skip possible optimizations and even cause failures on more complicated queries.

Reply:
Defaulted to false